### PR TITLE
comfyui: init at 0.15.1

### LINKED ISSUE #227006: Package request: comfyui
**Project description**

[ComfyUI](https://github.com/comfyanonymous/ComfyUI/) is a node editor for AI models such as stable diffusion. It requires a few python packages that are not currently packaged in nixpkgs. It also would be nice to get #226998 implemented first.

See https://github.com/comfyanonymous/ComfyUI/blob/master/requirements.txt for more information.

**Metadata**

* homepage URL: https://github.com/comfyanonymous/ComfyUI
* source URL: https://github.com/comfyanonymous/ComfyUI
* license: GPL v3
* platforms: linux, *maybe* darwin or other unix systems

An example of running comfyui with ipv6 support and **without** NixOS:
```sh
python3 -m pip install -r requirements.txt
python3 main.py --listen :: --cpu # specifies to use only the CPU with pytorch
```

### PR DESCRIPTION
**Pull request by 06kellyjac**
## Things done

closes #227006

Draft for now as undergoing testing and in an early stage.
Still missing some of the extra bits like `TRITON_LIBCUDA_PATH` and adjusting `LD_LIBRARY_PATH` seen in https://github.com/Janrupf/stable-diffusion-webui-nix which is probably necessary.

Went with `comfyui` rather than `comfy-ui` since the original ComfyUI isn't hyphenated. Also the sub-packages could have been hyphenated `comfy-ui` but weren't e.g. `comfyui-embedded-docs`.

The `comfyui-*` specific python modules could maybe live under the `comfyui` dir as they won't see any other use elsewhere, I don't mind.

- Built on platform:
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [ ] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by 06kellyjac** [comment]
should do, you're welcome to test it out.

Sometimes there are edge cases to handle with the dependencies for CUDA and ROCm stuff but for this particular package there's nothing split out in "optional"

I've got a module I'm playing with locally I'll push at some point.

**Comment by fillon** [comment]
is it still a WIP?

**Comment by ksf** [comment]
This is kinda OT but I wouldn't bet on generic rocm builds: Before I bought a new GPU I was stuck with an RX 5500, no official support, but with an environment variable you could get it to work by pretending its' another GPU. That only worked up to a specific rocm version as then AMD updated the kernels, and now they were using instructions that my card didn't have. Not wanting to be stuck on the old version I did shop around for a generic build, found one, and lo and behold it worked... and was like 10x slower. Those hand-rolled kernels really do a lot of the heavy lifting.

Haven't tried the new rocm builds on unstable yet (which would be required, 6.3 doesn't support gfx1200), I just downloaded torch+rocm via pip. Definitely going to try once it hits stable.

...oh and I actually upgraded my dev flake to a FHS env because there's pip dependencies which come with executables. It'd sure be nice to have everything managed by nix and I did try for a good while but as I like to try out new techniques (which means custom nodes) and noone upstream cares about, well, frankly speaking, software design principles, it's just not worth the bother. If it was a thing that needed fixing once, sure, but it's an uphill battle against a fleeing target.

As far as security is concerned: Just run it as a different user. It is a web app, after all.

**Comment by 06kellyjac** [comment]
I just saw ComfyUI Manager uses https://registry.comfy.org/ and wrote a little script to take the nodes and sort by top installed nodes.
Turns out ComfyUI-GGUF isn't even in the top 10 currently, it's 14th.


Here's the first 15
```json
[
  {
    "name": "ComfyUI-KJNodes",
    "total_install": 1486151,
    "github_stars": 1835,
    "status": "active",
    "latest_version": "1.1.7",
    "description": "Various quality of life -nodes for ComfyUI, mostly just visual stuff to improve usability.",
    "publisher_id": "kijai"
  },
  {
    "name": "rgthree-comfy",
    "total_install": 1325037,
    "github_stars": 2301,
    "status": "active",
    "latest_version": "1.0.2510052058",
    "description": "Making ComfyUI more comfortable.",
    "publisher_id": "rgthree"
  },
  {
    "name": "ComfyUI-Easy-Use",
    "total_install": 1285546,
    "github_stars": 1984,
    "status": "active",
    "latest_version": "1.3.4",
    "description": "To enhance the usability of ComfyUI, optimizations and integrations have been implemented for several commonly used nodes.",
    "publisher_id": "yolain"
  },
  {
    "name": "ComfyUI Impact Pack",
    "total_install": 1258247,
    "github_stars": 2723,
    "status": "active",
    "latest_version": "8.25.1",
    "description": "This node pack offers various detector nodes and detailer nodes that allow you to configure a workflow that automatically enhances facial details. And provide iterative upscaler.",
    "publisher_id": "drltdata"
  },
  {
    "name": "ComfyUI-VideoHelperSuite",
    "total_install": 1245433,
    "github_stars": 1246,
    "status": "active",
    "latest_version": "1.7.7",
    "description": "Nodes related to video workflows",
    "publisher_id": "kosinkadink"
  },
  {
    "name": "ComfyUI_essentials",
    "total_install": 1155765,
    "github_stars": 961,
    "status": "active",
    "latest_version": "1.1.0",
    "description": "Essential nodes that are weirdly missing from ComfyUI core. With few exceptions they are new features and not commodities.",
    "publisher_id": "matteo"
  },
  {
    "name": "ComfyUI-Custom-Scripts",
    "total_install": 1083650,
    "github_stars": 2732,
    "status": "active",
    "latest_version": "1.2.5",
    "description": "Enhancements & experiments for ComfyUI, mostly focusing on UI features",
    "publisher_id": "pythongosssss"
  },
  {
    "name": "comfyui_controlnet_aux",
    "total_install": 1024043,
    "github_stars": 3456,
    "status": "active",
    "latest_version": "1.1.2",
    "description": "Plug-and-play ComfyUI node sets for making ControlNet hint images",
    "publisher_id": "fannovel16"
  },
  {
    "name": "ComfyUI Impact Subpack",
    "total_install": 883943,
    "github_stars": 263,
    "status": "active",
    "latest_version": "1.3.5",
    "description": "This node pack provides nodes that complement the Impact Pack, such as the UltralyticsDetectorProvider.",
    "publisher_id": "drltdata"
  },
  {
    "name": "CG Use Everywhere",
    "total_install": 857972,
    "github_stars": 809,
    "status": "pending",
    "latest_version": "7.3.0",
    "description": "A set of nodes that allow data to be 'broadcast' to some or all unconnected inputs. Greatly reduces link spaghetti.",
    "publisher_id": "chrisgoringe"
  },
  {
    "name": "ComfyUI_LayerStyle",
    "total_install": 758234,
    "github_stars": 2617,
    "status": "pending",
    "latest_version": "1.0.90",
    "description": "A set of nodes for ComfyUI it generate image like Adobe Photoshop's Layer Style. the Drop Shadow is first completed node, and follow-up work is in progress.",
    "publisher_id": "chflame163"
  },
  {
    "name": "ComfyUI-WanVideoWrapper",
    "total_install": 738119,
    "github_stars": 4802,
    "status": "active",
    "latest_version": "1.3.6",
    "description": "ComfyUI wrapper nodes for WanVideo",
    "publisher_id": "kijai"
  },
  {
    "name": "ComfyUI-Florence2",
    "total_install": 729654,
    "github_stars": 1457,
    "status": "active",
    "latest_version": "1.0.6",
    "description": "Nodes to use Florence2 VLM for image vision tasks: object detection, captioning, segmentation and ocr",
    "publisher_id": "kijai"
  },
  {
    "name": "ComfyUI-GGUF",
    "total_install": 709519,
    "github_stars": 2645,
    "status": "active",
    "latest_version": "1.1.4",
    "description": "GGUF Quantization support for native ComfyUI models.",
    "publisher_id": "city96"
  },
  {
    "name": "ComfyUI-Frame-Interpolation",
    "total_install": 683387,
    "github_stars": 835,
    "status": "active",
    "latest_version": "1.0.7",
    "description": "A custom node suite for Video Frame Interpolation in ComfyUI",
    "publisher_id": "fannovel16"
  },
  # ...
]
```

File: pkgs/by-name/co/comfyui/nodes/comfyui-gguf.nix [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  version = "2.0.0-unstable-2026-01-12";
+
+  src = fetchFromGitHub {
+    owner = "city96";
+    repo = "ComfyUI-GGUF";
+    rev = "6ea2651e7df66d7585f6ffee804b20e92fb38b8a";
+    hash = "sha256-/ZwecgxTTMo9J1whdEJci8lEkOy/yP+UmjbpOAA3BvU=";
+  };
+
+  banditSkipChecks = [
```
Comment (SuperSandro2000) [inline pkgs/by-name/co/comfyui/nodes/comfyui-gguf.nix:19]: We definitely never want to run the bandit linter in our tests as it is an easy cause for breakages on bandit updates that are easily preventable.

**Comment by SuperSandro2000** [comment]
Going to close this in favor of #515744 as it is a lot more major already. Thanks for your help there! If anything is missing from there, feel free to request my review in another PR and I try to look at it within a few days.
